src/types.jl

```diff
@@ -1,9 +1,33 @@
 """
-`Compiled` is a trait indicating that any `:call` expressions should be evaluated
-using Julia's normal compiled-code evaluation. The alternative is to pass `stack=Frame[]`,
-which will cause all calls to be evaluated via the interpreter.
+    abstract type Interpreter end
+
+An interpreter that subtypes this type can implement its own evaluation strategies, by
+overloading the certain methods in JuliaInterpreter that are defined for this base type.
+The default behaviors of `Interpreter` is same as that of [`RecursiveInterpreter`](@ref),
+meaning it will recursively interpret all `:call` expressions.
+"""
+abstract type Interpreter end
+
 """
-struct Compiled end
+    RecursiveInterpreter <: Interpreter
+
+`RecursiveInterpreter` is an [`Interpreter`](@ref) that recursively interprets any `:call`
+expressions in the code being interpreted.
+"""
```

Comment:
I'm not 100% sure that this suggestion is correct, but if it is the case it might help unfamiliar readers such as myself. Feel free to accept or ignore.
```suggestion
`RecursiveInterpreter` is an [`Interpreter`](@ref) that recursively interprets any `:call`
expressions in the code being interpreted.

With this interpreter, code runs in fully interpreted mode; it will never be compiled for execution.
"""
```

Reply:
This additional comment would be very helpful:)